### PR TITLE
Simplify make build

### PR DESCRIPTION
**Pull request by Tombana**
## What do these changes do?
- This updates the `Makefile` with changes in the upstream Tensorflow `Makefile`.
- This simplifies the build a bit: `lce_register.cc` is removed as we now build our own `lce_benchmark` tool instead of building the TF native `benchmark` tool with our ops 'hacked' into it.
- This places the output binaries in `<repo>/gen` instead of a weird deep subfolder of the tensorflow submodule. (This is now also updated in https://github.com/larq/docs/pull/181)

## Related issue number
Closes https://github.com/larq/compute-engine/issues/408
Closes https://github.com/larq/compute-engine/issues/396
The `aarch64_aarch64` --> `linux_aarch64` path error in these issues seemed to be fixed by https://github.com/tensorflow/tensorflow/commit/69dc52a70472c84ca1f88262c8adea2bcd5e5809 and not by this PR, but I think this PR is a nice place to close these issues anyway.

### CODE REVIEW

File: larq_compute_engine/tflite/build_make/Makefile [hunk 1]
```diff
@@ -54,37 +58,69 @@ INCLUDES := \
...
 INCLUDES += -I/usr/local/include
 
+# These are the default libraries needed, but they can be added to or
+# overridden by the platform-specific settings in target makefiles.
+LIBS := \
+-lstdc++ \
+-lpthread \
+-lm \
+-lz \
+-ldl
```
Comment (Tombana) [inline larq_compute_engine/tflite/build_make/Makefile:68]: This is an update from the TF `Makefile`.

File: larq_compute_engine/tflite/build_make/Makefile [hunk 2]
```diff
@@ -54,37 +58,69 @@ INCLUDES := \
...
+
 # There are no rules for compiling objects for the host system (since we don't
 # generate things like the protobuf compiler that require that), so all of
 # these settings are for the target compiler.
-CFLAGS := -O3 -DNDEBUG -fPIC
+CFLAGS := -O3 -DNDEBUG -fPIC $(EXTRA_CFLAGS)
```
Comment (Tombana) [inline larq_compute_engine/tflite/build_make/Makefile:73]: This is an update from the TF `Makefile`.

File: larq_compute_engine/tflite/build_make/Makefile [hunk 3]
```diff
@@ -54,37 +58,69 @@ INCLUDES := \
...
+ifneq ($(shell cat /proc/cpuinfo | grep Features | grep neon),)
+  ifneq ($(shell cat /proc/cpuinfo | grep Features | grep vfpv4),)
+    CXXFLAGS += -mfpu=neon-vfpv4
+  else
+    CXXFLAGS += -mfpu=neon
+  endif
+endif # ifeq ($(TARGET_ARCH),armv7l)
+endif # ifeq ($(HOST_ARCH),$(TARGET_ARCH))
+endif # ifeq ($(HOST_OS),$(TARGET))
+endif
```
Comment (Tombana) [inline larq_compute_engine/tflite/build_make/Makefile:104]: This is an update from the TF `Makefile`.

**Review by Tombana — COMMENTED** [review]
I've tagged the changes that were coming from the upstream `Makefile`, for easier reviewing.

**Review by lgeiger — APPROVED** [review]
This is great! Thank you for the cleanup. This will also make it easier to upgrade TensorFlow versions since we don't need to keep `lce_register.cc` uptodate anymore.

**Comment by lgeiger** [comment]
I checked that this also works for natively building the binaries on macos 👍

**Review by AdamHillier — APPROVED** [review]
Awesome

**Review by lgeiger — APPROVED** [review]

**Pull request merged**